### PR TITLE
Update check_patroni for patroni 4.0.0

### PR DESCRIPTION
**Pull request by blogh**
In this version of patroni :

* Patroni role is written to DCS as primary instead of master.
* Patroni role returned by Patroni REST API has been changed from master to primary.

Update tests and documentation to reflect this.

### CODE REVIEW

File: tests/cluster_config_has_changed.state_file
Comment (dlax) [inline tests/cluster_config_has_changed.state_file:1]: Did you commit this file on purpose? I don't see where it's used.

File: tests/node_tl_has_changed.state_file
Comment (dlax) [inline tests/node_tl_has_changed.state_file:1]: Same for this one.

Comment (blogh) [reply tests/node_tl_has_changed.state_file:1]: you re right it's a mistake. thanks

Comment (blogh) [reply tests/node_tl_has_changed.state_file:1]: I have removed the state files, updated the `.gitignore` and fixed a typo in `vagrant/check_patroni.sh`.

**Review by dlax — APPROVED** [review]

**Comment by blogh** [comment]
thx @dlax

**Pull request merged**